src/anyio/_backends/_asyncio.py

```diff
@@ -751,12 +732,15 @@ async def __aexit__(
                 if not isinstance(exc_val, CancelledError):
                     self._exceptions.append(exc_val)
 
+            loop = get_running_loop()
             try:
                 if self._tasks:
                     with CancelScope() as wait_scope:
                         while self._tasks:
+                            if self._on_completed_fut is None:
+                                self._on_completed_fut = loop.create_future()
                             try:
```

Comment:
Add a blank line after a control block.
```suggestion
                                self._on_completed_fut = loop.create_future()

                            try:
```